doc/source/whatsnew/v0.23.0.txt

```diff
@@ -408,6 +408,8 @@ Indexing
 - Bug in :func:`MultiIndex.set_labels` which would cause casting (and potentially clipping) of the new labels if the ``level`` argument is not 0 or a list like [0, 1, ... ]  (:issue:`19057`)
 - Bug in ``str.extractall`` when there were no matches empty :class:`Index` was returned instead of appropriate :class:`MultiIndex` (:issue:`19034`)
 - Bug in :class:`IntervalIndex` where set operations that returned an empty ``IntervalIndex`` had the wrong dtype (:issue:`19101`)
+- Bug in :func:`DataFrame.drop`, ``KeyError`` now raises when dropping an ``Index`` or column that has duplicates (:issue:`19186`)
```

Comment:
add another entry in API breaking saysing that ``.drop()`` for ``Series,Index,DataFrame,Panel .drop`` now raises ``KeyError`` rather than ``ValueError`` if labels are missing (use this PR number)